### ISSUE TITLE
forge verify-contract with option flatten is failing

### ISSUE THREAD

**Issue by akoita** [foundry-rs/foundry#3365]
### Component

Forge

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (9e29032 2022-09-26T00:04:52.53004411Z)

### What command(s) is the bug in?

forge verify-contract --flatten 

### Operating System

Linux

### Describe the bug

Hi, 

I'm trying to verify my contract on a private instance of Blockscout (but same error when using Etherscan too) with a command like this: 

`forge verify-contract --num-of-optimizations 200 --flatten --force --watch --verifier blockscout --verifier-url <url> --compiler-version 0.8.13+commit.abaa5c0e --chain-id <id> --libraries <lib-path>:<lib-name>:<lib-address> --constructor-args 0x <contract-adress> <contract-path>:<contract-name>`

But I' m getting this error: 
```
2022-09-26T09:41:57.504644Z ERROR etherscan: Failed to deserialize response: invalid type: null, expected a string at line 1 column 48 response="{\"message\":\"Something went wrong.\",\"result\":null,\"status\":\"0\"}"
```

And the final error message with the payload:
```

Error:
Failed to submit contract verification, payload:
{"contractaddress":"0xe0a37e9bb40d04ae614535c60dcd69f1dc48cd13","sourceCode":"// SPDX-License-Identifier: MIT\npragma solidity 0.8.13;\n\nlibrary BoobaContractLibrary {\n    function helloMessage() public pure returns (string memory) {\n        return \"Bonjour le monde!\";\n    }\n}\n\ncontract BoobaContract {\n\n    constructor()  {}\n\n    function hello() public pure returns (string memory) {\n        return BoobaContractLibrary.helloMessage();\n    }\n}\n","codeformat":"solidity-single-file","contractname":"BoobaContract","compilerversion":"v0.8.13+commit.abaa5c0e","optimizationUsed":"1","runs":"200"}

Context:
- invalid type: null, expected a string at line 1 column 48
```

And it turns out that when I remove the option "flatten", the command runs successfully, but the contract remains unverified on Blockscout. 

Any idea why I have this error? 

Thanks

**Comment by mattsse** [comment]
there's another blockscout verify issue #3331 

investigating now.

could you please check whether the output of `forge flatten ...` is valid solidity, flatten has a hard time with renames for example.

**Comment by akoita** [comment]
Oh sorry, I missed the other issue! Otherwise I have the same error using Etherscan too ( Goerli instance)

OK, I'll check the output of the `flatten` command. But I had already used the flattened contract (~4000 rows) to successfully check the contract manually in Blockscout UI a few weeks ago.

I just saw that the problem with #3331 was the use of an incorrect url, so here are the urls I tried:

- Blockscout: https://api-goerli.etherscan.io
- Etherscan: https://api-goerli.etherscan.io
